modules/file-system/filestore/main.tf

```diff
@@ -79,4 +78,19 @@ resource "google_filestore_instance" "filestore_instance" {
       delete = "1h"
     }
   }
+
+  lifecycle {
+    precondition {
+      condition = (
+        var.reserved_ip_range == null ||
+        var.connect_mode == "DIRECT_PEERING" && can(cidrhost(var.reserved_ip_range, 0)) && contains(["24", "29"], try(split("/", var.reserved_ip_range)[1], "")) ||
+        var.connect_mode == "PRIVATE_SERVICE_ACCESS"
```

Comment:
nit: move the `PRIVATE_SERVICE_ACCESS` condition above the `DIRECT_PEERING` condition. It will have properties of return early clause and will make it more readable.

Reply:
I can do that. Terraform does not short-circuit logic, though, if that's what you mean.

Reply:
I was mostly thinking for readability.